File: library/src/main/java/org/incendo/interfaces/click/ClickContext.java
package org.incendo.interfaces.click;

import org.bukkit.entity.Player;
import org.bukkit.event.inventory.ClickType;
import org.incendo.interfaces.view.InterfaceView;

public record ClickContext(
    Player player,
    InterfaceView view,
    ClickType type
) {
}

File: next/src/main/java/org/incendo/interfaces/next/click/ClickContext.kt
package org.incendo.interfaces.next.click

import org.bukkit.entity.Player
import org.bukkit.event.inventory.ClickType
import org.incendo.interfaces.next.view.AbstractInterfaceView

public data class ClickContext(
    public val player: Player,
    public val view: AbstractInterfaceView<*, *>,
    public val type: ClickType
)
